DarwinWorld/src/main/java/agh/ics/oop/model/maps/GlobeMap.java

```diff
@@ -64,20 +70,39 @@ public void reproduceAnimals() {
             }
         }
     }
-    private List<Integer> calculatePrefferedRows(int width, int height)  {
+    public void growPlants() {
+        Random rand = new Random();
+        for(int i=0; i<config.getDailyPlantsGrowth(); i++) {
+            boolean choosePreferredField = rand.nextInt(5) != 0;
+            if(choosePreferredField && emptyPreferredFields.size()>0) {
+                findRandomFieldAndGrow(emptyPreferredFields);
+            } else {
+                if(emptyNormalFields.size() > 0) {
+                    findRandomFieldAndGrow(emptyNormalFields);
+                }
+            }
+        }
+    }
+    private void findRandomFieldAndGrow(List<MapField> fields) {
+        Random rand = new Random();
+        int randomIndex = rand.nextInt(fields.size());
+        MapField chosenField = fields.remove(randomIndex);
+        chosenField.growPlant();
+    }
+    private List<Integer> calculatePreferredRows(int width, int height)  {
             int fieldsAmount = width*height;
-            List<Integer> prefferedRows = new LinkedList<>();
+            List<Integer> preferredRows = new LinkedList<>();
             int centralRow = height/2;
             int side = -1;
             int currentRow=centralRow;
-            prefferedRows.add(centralRow);
-            while(prefferedRows.size()*width < fieldsAmount*0.2) {
-                int newRow = currentRow+side*prefferedRows.size();
-                prefferedRows.add(newRow);
+            preferredRows.add(centralRow);
+            while(preferredRows.size()*width < fieldsAmount*0.2) {
```

Comment:
0.2 powinno mieć swoją zmienną bo tak to nie wiadomo o co chodzi
float preferredFieldsCoveragePercentage = 0.2f